source/common/formatter/substitution_formatter.cc

```diff
@@ -1440,6 +1440,9 @@ const StreamInfoFormatter::FieldExtractorLookupTbl& StreamInfoFormatter::getKnow
                                                    .get()
                                                    .upstreamTransportFailureReason();
                                     }
+                                    if (result) {
+                                      std::replace(result->begin(), result->end(), ' ', '_');
```

Comment:
Is this needed? IIRC we used to have something like this, and decided to remove it and just have all the code correctly format the string to not have spaces.

Reply:
We could do either a) Format all existing strings to remove space. This will probably require us to do a verification of presence of spaces when the field is set by new code path b) Replace all spaces (as implemented here). I think b) is easier and limits the scope of this PR but I am open to suggestion.